lib/native/eventstream.ts

```diff
@@ -835,6 +871,15 @@ export class ClientStream extends NativeResourceMixin(BufferedEventEmitter) {
         ));
     }
 
+    /**
+     * Shuts down the stream (if active) and begins the process to release native resources associated with it by
+     * having the native binding release the only reference to the extern object representing the stream.
+     *
+     * Ultimately, the native resources will not be released until the native stream has fully shut down and that
+     * shutdown event has reached the libuv event loop.
+     *
+     * This function **must** be called for every ClientStream instance or native resources will leak.
```

Comment:
Minor, debatable: I'd add something like `... **must be called for every ClientStream instance when finished, or native resources will leak` just to further highlight that you should only call `close()` when fully finished and not planning to use it anymore. Not sure if it is totally needed though since it's mentioned in other callbacks and in the class definition itself...